scalding-core/src/main/scala/com/twitter/scalding/serialization/StringOrderedSerialization.scala

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.serialization
+
+import java.io.{ InputStream, OutputStream }
+import scala.util.{ Failure, Success, Try }
+import scala.util.control.NonFatal
+
+import JavaStreamEnrichments._
+
+object StringOrderedSerialization {
+  final def binaryIntCompare(leftSize: Int, seekingLeft: InputStream, rightSize: Int, seekingRight: InputStream): Int = {
```

Comment:
This works because comparing the binary of utf-8 as unsigned bytes gives the same sort as comparing the unicode code-points by index, which is what String.compare does. This is only coupled to the encoding in that the serialized data was written with UTF-8, but we are doing the reading and writing. We can add some comments.